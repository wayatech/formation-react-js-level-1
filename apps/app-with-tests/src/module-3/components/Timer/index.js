import React from "react";
import PropTypes from "prop-types";

class Timer extends React.Component {
  constructor(props) {
    super(props);
    this.state = { seconds: 0 };
  }

  componentDidMount() {
    this.interval = setInterval(() => this.tick(), 1000);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  tick() {
    this.setState((state) => ({
      seconds: state.seconds + 1,
    }));
  }

  render() {
    const { bgColor } = this.props;
    const { seconds } = this.state;

    return <p style={{ backgroundColor: bgColor }}>Secondes : {seconds}</p>;
  }
}

Timer.propTypes = {
  bgColor: PropTypes.string.isRequired,
};

export default Timer;
