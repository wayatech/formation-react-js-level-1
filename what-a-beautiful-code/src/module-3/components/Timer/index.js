import React from 'react'
import PropTypes from 'prop-types'

class Timer extends React.Component {
  constructor(props) {
    super(props)
    this.state = { seconds: 0 }
  }

  componentDidMount() {
    this.interval = setInterval(() => this.tick(), 1000)
  }

  componentWillUnmount() {
    clearInterval(this.interval)
  }

  tick() {
    this.setState((state) => ({
      seconds: state.seconds + 1,
    }))
  }

  render() {
    const { bgColor } = this.props
    const { seconds } = this.state

    return <div style={{ backgroundColor: bgColor }}>Secondes : {seconds}</div>
  }
}

Timer.propTypes = {
  bgColor: PropTypes.func.isRequired,
}

export default Timer
