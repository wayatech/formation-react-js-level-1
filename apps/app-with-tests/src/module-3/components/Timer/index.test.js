import React from "react";
import renderer from "react-test-renderer";

import Timer from "./index";

test("Timer display correctly", () => {
  const component = renderer.create(<Timer bgColor="yellow" />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
