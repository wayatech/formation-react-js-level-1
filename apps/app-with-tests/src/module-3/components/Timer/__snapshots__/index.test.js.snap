// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer display correctly 1`] = `
<p
  style={
    Object {
      "backgroundColor": "yellow",
    }
  }
>
  Secondes : 
  0
</p>
`;
